velox/exec/tests/SortBufferTest.cpp

```diff
@@ -611,13 +611,66 @@ DEBUG_ONLY_TEST_F(SortBufferTest, reserveMemoryGetOutput) {
     // Sets an extreme large value to get output once to avoid test flakiness.
     sortBuffer->getOutput(1'000'000);
     if (spillEnabled) {
-      ASSERT_EQ(numReserves, 1);
+      // Reserve memory for sort and getOutput.
+      ASSERT_EQ(numReserves, 2);
     } else {
       ASSERT_EQ(numReserves, 0);
     }
   }
 }
 
+DEBUG_ONLY_TEST_F(SortBufferTest, reserveMemorySort) {
+  for (bool usePrefixSort : {false, true}) {
+    for (bool spillEnabled : {false, true}) {
+      SCOPED_TRACE(fmt::format("spillEnabled {}", spillEnabled));
+      auto spillDirectory = exec::test::TempDirectoryPath::create();
+      auto spillConfig = getSpillConfig(spillDirectory->getPath());
+      folly::Synchronized<common::SpillStats> spillStats;
+      const RowTypePtr inputType = usePrefixSort ? inputType_
+                                                 : ROW(
+                                                       {{"c0", VARCHAR()},
+                                                        {"c1", VARCHAR()},
+                                                        {"c2", VARCHAR()},
+                                                        {"c3", VARCHAR()},
+                                                        {"c4", VARCHAR()},
+                                                        {"c5", VARCHAR()}});
+      auto sortBuffer = std::make_unique<SortBuffer>(
+          inputType,
+          sortColumnIndices_,
+          sortCompareFlags_,
+          pool_.get(),
+          &nonReclaimableSection_,
+          prefixSortConfig_,
+          spillEnabled ? &spillConfig : nullptr,
+          &spillStats);
+
+      const std::shared_ptr<memory::MemoryPool> spillSource =
+          memory::memoryManager()->addLeafPool("spillSource");
+      VectorFuzzer fuzzer({.vectorSize = 100}, spillSource.get());
+
+      TestScopedSpillInjection scopedSpillInjection(0);
+      sortBuffer->addInput(fuzzer.fuzzRow(inputType));
+
+      std::atomic_bool hasReserveMemory = false;
+      // Reserve memory for sort.
+      SCOPED_TESTVALUE_SET(
+          "facebook::velox::common::memory::MemoryPoolImpl::maybeReserve",
+          std::function<void(memory::MemoryPoolImpl*)>(
+              ([&](memory::MemoryPoolImpl* pool) {
+                hasReserveMemory.store(true);
+              })));
+
+      sortBuffer->noMoreInput();
+      if (spillEnabled) {
```

Comment:
Shall we check the combination of spillEnabled and usePrefixSort? And we expect hasReserveMemory-> false if spillEnabled && !usePrefixSort? Thanks!